asktext.py

```diff
@@ -213,6 +217,29 @@ def __init__(self, text, preamble_file, global_scale_factor, current_scale_facto
             self._frame = None
             self._scale = None
 
+        @staticmethod
+        def validate_spinbox_input(d, i, P, s, S, v, V, W):
+            """ Ensure that only floating point numbers are accepted as input of a Tk widget
+                Inspired from:
+                https://stackoverflow.com/questions/4140437/interactively-validating-entry-widget-content-in-tkinter
+
+                Note: Selecting an entry and copying something from the clipboard is rejected by this method. Reason:
+                Without validation Tk appends the content of the clipboard at the end of the selection leading to
+                invalid content. So with or without this validation method you need to delete the content of the
+                box before inserting the new stuff.
+            """
+            if S == '' or P == '':
+                # Initialization of widget (S=='') and deletion of entry (P=='')
+                valid = True
+            else:
+                # All other cases: Ensure that result is OK
+                try:
+                    float(P)
```

Comment:
interesting trick?

Reply:
S is the input coming from the user (a character from the keyboard or the content of the clipboard), s is the string currently in the box and P is the string which will be placed into the box after the input process ended. P is set by Tk.  

Content of box: s="1.23", User input: S=","  --> P = "1.23,"  --> Input rejected
Content of box s="1.23", User input: S="5.44" from clipboard --> P = "1.235.44" !! [Even without the validation method I had to delete the entry before entering a new one from clipboard. Selected entries are not overwritten as one would naturally expect. Strange Tk.]

Will document it.

Reply:
I specifically meant the `float(P)` part to validate the input.